neural_lam/weather_dataset.py

```diff
@@ -94,6 +125,98 @@ def __init__(
+        else:
+            self.da_state = self.da_state
+
+        # Check time step consistency in state data
+        if self.datastore.is_forecast:
+            state_times = self.da_state.analysis_time
+            self.forecast_step_state = self._get_time_step(
+                self.da_state.elapsed_forecast_duration
+            )
+        else:
+            state_times = self.da_state.time
+        self.time_step_state = self._get_time_step(state_times)
+
+        # Check time coverage for forcing and boundary data
+        if self.da_forcing is not None or self.da_boundary is not None:
+            if self.datastore.is_forecast:
+                state_times = self.da_state.analysis_time
+            else:
+                state_times = self.da_state.time
+            state_time_min = state_times.min().values
+            state_time_max = state_times.max().values
+
+            if self.da_forcing is not None:
+                # Forcing data is part of the same datastore as state data
+                # During creation the time dimension of the forcing data
+                # is matched to the state data
+                if self.datastore.is_forecast:
+                    forcing_times = self.da_forcing.analysis_time
+                    self.forecast_step_forcing = self._get_time_step(
+                        self.da_forcing.elapsed_forecast_duration
+                    )
+                else:
+                    forcing_times = self.da_forcing.time
+                self.time_step_forcing = self._get_time_step(
+                    forcing_times.values
+                )
+
+            if self.da_boundary is not None:
+                # Boundary data is part of a separate datastore
+                # The boundary data is allowed to have a different time_step
+                # Check that the boundary data covers the required time range
+                if self.datastore_boundary.is_forecast:
+                    boundary_times = self.da_boundary.analysis_time
+                    self.forecast_step_boundary = self._get_time_step(
+                        self.da_boundary.elapsed_forecast_duration
+                    )
+                else:
+                    boundary_times = self.da_boundary.time
+                self.time_step_boundary = self._get_time_step(
+                    boundary_times.values
+                )
+                boundary_time_min = boundary_times.min().values
+                boundary_time_max = boundary_times.max().values
+
+                # Calculate required bounds for boundary using its time step
+                boundary_required_time_min = (
+                    state_time_min
+                    - self.num_past_forcing_steps * self.time_step_boundary
+                )
+                boundary_required_time_max = (
+                    state_time_max
+                    + self.num_future_forcing_steps * self.time_step_boundary
+                )
```

Comment:
I think there is a typo here w.r.t. which variables to use

```suggestion
                boundary_required_time_min = (
                    state_time_min
                    - self.num_past_boundary_steps * self.time_step_boundary
                )
                boundary_required_time_max = (
                    state_time_max
                    + self.num_future_boundary_steps * self.time_step_boundary
                )
```

Reply:
OOps, thank you :pray: I have in the meantime defined a new function that takes over the min/max time calculations: https://github.com/sadamov/neural-lam/blob/c82d22ba9749c8f1fcd04c7bd7e1301881951622/neural_lam/weather_dataset.py#L180 
This issue should be fixed